share/wake/lib/system/remote_cache_api.wake

```diff
@@ -533,78 +533,88 @@ export target rscApiCheckAuthorization (api: RemoteCacheApi): Result Unit Error
         Some x -> failWithError "Invalid auth key. Status code: {str x}"
         None -> failWithError "Invalid auth key. Unable to determine status code"
 
-# rscApiGetStringBlob: Downloads a blob and returns the contents as a string 
-#
-# ```
-#  rscApiGetStringBlob (RemoteCacheBlob "asdf" "https://...") = Pass "foo\nbar\nbat" 
-# ```
-export def rscApiGetStringBlob ((CacheSearchBlob _ uri): CacheSearchBlob): Result String Error =
-    require scheme, path, Nil = extract `([a-zA-Z][a-zA-Z0-9+.-]*)://(.*)` uri
-    else failWithError "rsc: uri has unexpected format: '{uri}'"
-
-    # maps path = "%46%6F%6F" to content = "foo"
-    def dbScheme _scheme path =
-        # If path is empty we have the empty string
-        require True = path.strlen > 0
-        else Pass ""
-
-        # If not empty then it must as least be %00
-        require True = path.strlen >= 3
-        else failWithError "rsc: Invalid db path: '{path}'"
-
-        require "", byteStrs = tokenize `%` path
-        else failWithError "rsc: Failed to extract bytes from db path: '{path}'"
-
-        require Some bytes =
-            byteStrs
-            | map (intbase 16)
-            | findNone
-        else failWithError "rsc: Failed to parse bytes strings into bytes from db path: '{path}'"
-
-        bytes
-        | map integerToByte
-        | cat
-        | Pass
-
-    def fileScheme scheme path =
-        require Pass response =
-            buildHttpRequest "{scheme}://{path}"
-            | setMethod HttpMethodGet
-            | makeRequest
+# Converts db only blob uris into actual data
+# maps path = "%46%6F%6F" to content = "foo"
+def resolveDbScheme _scheme path =
+    # If path is empty we have the empty string
+    require True = path.strlen > 0
+    else Pass ""
+
+    # If not empty then it must as least be %00
+    require True = path.strlen >= 3
+    else failWithError "rsc: Invalid db path: '{path}'"
+
+    require "", byteStrs = tokenize `%` path
+    else failWithError "rsc: Failed to extract bytes from db path: '{path}'"
+
+    require Some bytes =
+        byteStrs
+        | map (intbase 16)
+        | findNone
+    else failWithError "rsc: Failed to parse bytes strings into bytes from db path: '{path}'"
+
+    bytes
+    | map integerToByte
+    | cat
+    | Pass
 
-        Pass response.getHttpResponseBody
+# Dispatches the content resolution for uri based on its scheme
+def resolveUriResponse (dbSchemeFn: (String => String => Result a Error)) (fileSchemeFn: (String => String => Result a Error)) (uri: String) =
+    require scheme, path, Nil = (uri | extract `([a-zA-Z][a-zA-Z0-9+.-]*)://(.*)`)
```

Comment:
There's a couple places where this formation is used, but it feels like using the function as normal might be easier to read than the parenthesized piping.
```suggestion
    require scheme, path, Nil = extract `([a-zA-Z][a-zA-Z0-9+.-]*)://(.*)` uri
```